### PR TITLE
transaction manager tests

### PR DESCRIPTION
**Pull request by graetzer**
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/5516/

### CODE REVIEW

File: arangod/StorageEngine/TransactionState.cpp [hunk 1]
```diff
@@ -286,24 +286,13 @@ TransactionCollection* TransactionState::findCollection(TRI_voc_cid_t cid,
...
-    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL,
-                                   "cannot make a write transaction read-only");
-  }
-  if (AccessMode::isWriteOrExclusive(type) && AccessMode::isRead(_type) &&
-      _status != transaction::Status::CREATED) {
+void TransactionState::setExclusiveAccessType() {
+  if (_status != transaction::Status::CREATED) {
     THROW_ARANGO_EXCEPTION_MESSAGE(
         TRI_ERROR_INTERNAL,
         "cannot make a running read transaction a write transaction");
```
Comment (jsteemann) [inline arangod/StorageEngine/TransactionState.cpp]:
```suggestion
        "cannot change the type of a running transaction");
```

File: tests/Mocks/StorageEngineMock.cpp [hunk 2]
```diff
@@ -1431,7 +1431,7 @@ arangodb::Result TransactionStateMock::abortTransaction(arangodb::transaction::M
 }
 
 arangodb::Result TransactionStateMock::beginTransaction(arangodb::transaction::Hints hints) {
-  static std::atomic<TRI_voc_tid_t> lastId(0);
+//  static std::atomic<TRI_voc_tid_t> lastId(0);
```
Comment (jsteemann) [inline tests/Mocks/StorageEngineMock.cpp]:
```suggestion
```

File: tests/Mocks/StorageEngineMock.cpp [hunk 3]
```diff
@@ -1445,18 +1445,22 @@ arangodb::Result TransactionStateMock::beginTransaction(arangodb::transaction::H
     return res;
   }
 
-  const_cast<TRI_voc_tid_t&>(_id) = ++lastId;  // ensure each transaction state has a unique ID
-  updateStatus(arangodb::transaction::Status::RUNNING);
+  if (nestingLevel() == 0) {
+//    const_cast<TRI_voc_tid_t&>(_id) = ++lastId;  // ensure each transaction state has a unique ID
```
Comment (jsteemann) [inline tests/Mocks/StorageEngineMock.cpp]:
```suggestion
```

File: tests/Mocks/StorageEngineMock.cpp [hunk 4]
```diff
@@ -1445,18 +1445,22 @@ arangodb::Result TransactionStateMock::beginTransaction(arangodb::transaction::H
...
   ++commitTransactionCount;
-  updateStatus(arangodb::transaction::Status::COMMITTED);
+  if (nestingLevel() == 0) {
+    updateStatus(arangodb::transaction::Status::COMMITTED);
+    // avoid use of TransactionManagerFeature::manager()->unregisterTransaction(...)
+    const_cast<TRI_voc_tid_t&>(_id) = 0;
+  }
   unuseCollections(nestingLevel());
-  // avoid use of TransactionManagerFeature::manager()->unregisterTransaction(...)
-  const_cast<TRI_voc_tid_t&>(_id) = 0;
```
Comment (jsteemann) [inline tests/Mocks/StorageEngineMock.cpp]: :+1:

File: CHANGELOG [hunk 5]
```diff
@@ -5,6 +5,9 @@ devel
 
 * added collection.documentId method to derive document id from key
 
+* Indexes can be created with the 'inBackground' will be created without an
```
Comment (danielhlarkin) [inline CHANGELOG]: "Indexes created with the..."?

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/5523/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/5545/

**Pull request merged**